cogs/menudocs.py

```diff
@@ -97,6 +88,62 @@ async def on_thread_join(self, thread) -> None:
 
         await thread.join()
 
+    @commands.Cog.listener()
+    async def on_member_update(self, before: disnake.Member, after: disnake.Member):
+        if after.guild.id != MAIN_GUILD:
+            # Not in menudocs
+            return
+
+        if before.roles == after.roles:
+            # If roles are the same
+            return
+
+        welcome_messages = [
+            "$MEMBER just showed up. Hold my beer.",
+            "Where's $MEMBER? In the server!",
+            "Welcome, $MEMBER. Stay awhile and listen.",
+            "We've been expecting you, $MEMBER.",
+            "$MEMBER just joined. Can I get a heal?",
+            "$MEMBER just joined the server - glhf!",
+            "Ermagherd. $MEMBER has joined us!",
+            "Mission Control! $MEMBER has successfully landed!"
+        ]
+
+        projections = after.guild.get_channel(MENUDOCS_PROJECTIONS_CHANNEL)
+        unverified_role = after.guild.get_role(MENUDOCS_UNVERIFIED_ROLE)
+        message = Template(random.choice(welcome_messages)).safe_substitute(
+            {"MEMBER": f"**{after}**"}
+        )
+        embed = disnake.Embed(description=message, colour=disnake.Colour.green())
+
+        if unverified_role in before.roles and not after.roles:
+            await projections.send(embed=embed)
+
+    @commands.Cog.listener()
+    async def on_member_remove(self, member: disnake.Member) -> None:
+        if member.guild.id != MAIN_GUILD:
+            # Not in menudocs
+            return
+
+        leave_messages  = [
+            "$MEMBER has quit. Party's over.",
+            "Ermagherd. $MEMBER has just left us here.",
+            "Brace yourselves. $MEMBER just abandoned the server.",
+            "$MEMBER just left. Can you come back?",
+            "Whoopsies! $MEMBER left us!",
+            "Nooooooo, $MEMBER closed the door.",
+            "Commander, we've lost $MEMBER!",
+            "Is this a loss? $MEMBER left."
+        ]
+
+        message = Template(random.choice(leave_messages)).safe_substitute(
+            {"MEMBER": f"**{member}**"}
+        )
+        projections = member.guild.get_channel(1003987467612991508)
```

Comment:
Is this also not the projections variable we already have?

Reply:
Oh yeah, that's my channel whoops.